### ISSUE TITLE
Improve label out of bounds error in CPU-mode F.softmax_cross_entropy

### ISSUE THREAD

**Issue by niboshi** [chainer/chainer#5551]
This is a suggestion only applicable in CPU mode.

If the maximum value of the label exceeds the number of data dimensions,
`IndexError` will be raised at this line:

https://github.com/chainer/chainer/blob/75b238f40618340de5f3c400fedaab51c6928803/chainer/functions/loss/softmax_cross_entropy.py#L98

We can catch the error and provide more informative error.

(related (in japanese): https://groups.google.com/forum/#!topic/chainer-jp/ks3Vfvx29aU)
